### PR TITLE
Additions to cloudbackup enumerate api (#1802)

### PR DESCRIPTION
**Pull request by talakad**
* Additions to cloudbackup enumerate api

Added cluster and namespace details to CloudBackupInfo structure

Signed-off-by: veda <veda@portworx.com>

* Review comments

Signed-off-by: veda <veda@portworx.com>

Notes: Cherry-pick from master

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -405,8 +405,16 @@ type CloudBackupGenericRequest struct {
 	// MetadataFilter indicates backups whose metadata has these kv pairs
 	MetadataFilter map[string]string
 	// CloudBackupID must be specified if one needs to enumerate known single
-	// backup( format is clusteruuidORBucketName/srcVolId-SnapId(-incr)
+	// backup (format is clusteruuidORBucketName/srcVolId-SnapId(-incr). If t\
```
Comment (lpabon) [inline api/api.go:408]: There is a `t\` at the end of the this comment

File: api/api.proto [hunk 2]
```diff
@@ -4239,6 +4242,21 @@ message SdkCloudBackupInfo {
   map<string, string> metadata = 5;
   // Status indicates the status of the backup
   SdkCloudBackupStatusType status = 6;
+  // cluster indicates if the cloudbackup belongs to current cluster,
```
Comment (lpabon) [inline api/api.proto:4245]: I think you can remove the word `cluster` and just start with `Indicates...`

File: api/api.proto [hunk 3]
```diff
@@ -4239,6 +4242,21 @@ message SdkCloudBackupInfo {
...
+  // with older cluster this value may be unknown
+  SdkCloudBackupClusterType cluster_type = 7;
+  // k8s namespace to which this backup belongs to
+  string namespace = 8; 
+}
+
+// CloudBackup operations types
+enum SdkCloudBackupClusterType {
```
Comment (lpabon) [inline api/api.proto:4253]: Enum is not complient with the SDK. Please see https://github.com/libopenstorage/openstorage/blob/master/docs/dev-sdk.md#enums

**Review by lpabon — CHANGES_REQUESTED** [review]
Looks good, just one changed is needed in the enum

File: api/api.go (hunk 1 above)
Comment (talakad) [reply api/api.go]: thanks!. done

**Comment by talakad** [comment]
Thank you @lpabon!

**Pull request merged**